### ISSUE TITLE
Button Remapping

### ISSUE THREAD

**Issue by ShadauxCat** [ShadauxCat/CATSFC#8]
I will be adding the option to re-map the controls if desired.  Along with this, I'll add the ability to map buttons to emulator functions when they aren't used in-game - for example, the L and R buttons can be used to scroll the screen instead of the stylus.  The button map settings will be game-specific.  Additionally, any button or setting will be mappable to a combination of buttons instead of a single button, so, for example, "Toggle Rapid Fire A" could be mapped to "L+R+A," or "Quick Save" could be mapped to "L+R+A+B+X+Y".

**Comment by Nebuleon** [comment]
Seeing as there is a perfect match between SNES controller buttons and DS controller buttons, I will not implement button remapping. By that, I mean that all buttons on the SNES controller are present on the DS, and all buttons on the DS controller are present on the SNES, at similar locations except for Start and Select.

I may implement global hotkeys and per-game hotkeys, though.

**Comment by ShadauxCat** [comment]
The main drive for me to implement this was the request that was made for L and R to scroll the screen... which then necessitated an ability to map other things to L and R. This is your project now so I'm not telling you what to do, just wanted to let you know _why_ I had put this in as a task.

**Comment by Nebuleon** [comment]
That then necessitates an ability to map buttons to be the buttons you chose for L and R, map buttons to those 2 last lost buttons and so on, which in effect requires you to lose the ability to use 2 keys.

In the DSTwo's DS game mode, keys that are part of the Real-Time Menu hotkey are still sent to the game before they take on their function as the Real-Time Menu hotkey. So, in _Super Mario World_, you'd pan the camera left and right with L and R, but you could assign R to a hypothetical Temporary Fast-Forward hotkey and it would fast forward **and** pan the camera, except in a castle destroyed sequence that you could fast-forward without invoking any function.

**Comment by ShadauxCat** [comment]
Yeah, but it could still be useful in some games. Some games leave certain buttons unused, or used so little that they can be thrown out in favor of the ability to move the screen. Ultimately I see no harm in letting the user do what they want, if it improves the overall experience for them.

Of course the other option would be to put buttons on the bottom screen to move the top screen up and down.

Then again, your anti-aliased mode from what I've seen in the screenshots seems to solve the problem.
